src/index.js

```diff
@@ -27,52 +28,63 @@ if (debugMode) {
   // output.start(1000 / 30)
 
   // bridge.setOutput(output)
-
+  
   // Use the actual DMX controller connected via USB
 } else {
   // Find all connected USB devices
-  Promise.resolve().then(() => {
-    if (usbDevicePort) {
-      console.log('Using device port from config.')
-      
-      return usbDevicePort
-    }
-    
-    return SerialPort.list().then(ports => {
-        return ports.find(usbPort => {
-          if (usbPort.comName.includes('usb')) {
-            console.log(usbPort)
-
-            // Find enttec
-            if (manufacturer.test(usbPort.manufacturer)) {
-              console.log('Found enttec DMX USB Pro')
-
-              return usbPort.comName
-            }
+  Promise.resolve()
+    .then(() => {
+      if (config.artNetDeviceAddress) {
+        return Promise.resolve().then(() => new ArtNetDriver(config.artNetDeviceAddress))
+      }
+
+      return Promise.resolve()
+        .then(() => {
+          if (usbDevicePort) {
+            console.log('Using device port from config.')
+
+            return usbDevicePort
+          }
+
+          return SerialPort.list().then(ports => ports.find(usbPort => {
+              if (usbPort.comName.includes('usb')) {
+                console.log(usbPort)
+
+                // Find enttec
+                if (manufacturer.test(usbPort.manufacturer)) {
+                  console.log('Found enttec DMX USB Pro')
+
+                  return usbPort.comName
+                }
+              }
+
+              return false
+            }))
+        })
+        .then(usbPort => {
+          if (!usbPort) {
+            console.error('No valid USB device found.')
+
+            return false
           }
+          // Use the port to find the correct controller automatcially
+          const usbproSerialport = new SerialPort(usbPort)
 
-          return false
+          return new EnttecUsbProMk2Driver(usbproSerialport)
         })
-      })
-  })
-  .then(usbPort => {
-      if (!usbPort) {
-        console.error('No valid USB device found.')
+    })
+    .then(driver => {
+      if (!driver) {
+        console.error('No driver was created.')
 
         return
       }
-      // Use the port to find the correct controller automatcially
-      const usbproSerialport = new SerialPort(usbPort)
-
-      const output = new DmxOutput(
-        new EnttecUsbProMk2Driver(usbproSerialport),
-        1
-      )
+      const output = new DmxOutput(driver, 1)
       output.start(1000 / 30)
 
       bridge.setOutput(output)
 
       console.log('Set enttec DMX USB Pro as the output for fivetwelve')
```

Comment:
This should be moved into line 72, as it is only related to the enttec controller, but not for the general driver.

Reply:
That should do it.